launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java

```diff
@@ -8,196 +8,65 @@
+    protected AbstractCommandBuilder(final Path wildflyHome, final Jvm jvm, final String moduleName) {
+        super(wildflyHome, jvm, moduleName);
     }
 
-    /**
-     * Sets whether or not the security manager option, {@code -secmgr}, should be used.
-     *
-     * @param useSecMgr {@code true} to use the a security manager, otherwise {@code false}
-     *
-     * @return the builder
-     */
+    @Override
     public T setUseSecurityManager(final boolean useSecMgr) {
-        this.useSecMgr = useSecMgr;
+        super.setUseSecurityManager(useSecMgr);
         return getThis();
     }
 
-    /**
-     * Indicates whether or no a security manager should be used for the server launched.
-     *
-     * @return {@code true} if a security manager should be used, otherwise {@code false}
-     */
-    public boolean useSecurityManager() {
-        return useSecMgr;
-    }
-
-    /**
-     * Adds a directory to the collection of module paths.
-     *
-     * @param moduleDir the module directory to add
-     *
-     * @return the builder
-     *
-     * @throws java.lang.IllegalArgumentException if the path is {@code null}
-     */
+    @Override
     public T addModuleDir(final String moduleDir) {
-        environment.addModuleDir(moduleDir);
+        super.addModuleDir(moduleDir);
         return getThis();
     }
 
-    /**
-     * Adds all the module directories to the collection of module paths.
-     *
-     * @param moduleDirs an array of module paths to add
-     *
-     * @return the builder
-     *
-     * @throws java.lang.IllegalArgumentException if any of the module paths are invalid or {@code null}
-     */
+    @Override
     public T addModuleDirs(final String... moduleDirs) {
-        environment.addModuleDirs(moduleDirs);
+        super.addModuleDirs(moduleDirs);
         return getThis();
     }
 
-    /**
-     * Adds all the module directories to the collection of module paths.
-     *
-     * @param moduleDirs a collection of module paths to add
-     *
-     * @return the builder
-     *
-     * @throws java.lang.IllegalArgumentException if any of the module paths are invalid or {@code null}
-     */
+    @Override
     public T addModuleDirs(final Iterable<String> moduleDirs) {
-        environment.addModuleDirs(moduleDirs);
+        super.addModuleDirs(moduleDirs);
         return getThis();
     }
 
-    /**
-     * Replaces any previously set module directories with the collection of module directories.
-     * <p/>
-     * The default module directory will <i>NOT</i> be used if this method is invoked.
-     *
-     * @param moduleDirs the collection of module directories to use
-     *
-     * @return the builder
-     *
-     * @throws java.lang.IllegalArgumentException if any of the module paths are invalid or {@code null}
-     */
+    @Override
     public T setModuleDirs(final Iterable<String> moduleDirs) {
-        environment.setModuleDirs(moduleDirs);
+        super.addModuleDirs(moduleDirs);
```

Comment:
This should be `setModuleDirs()` instead of the add variant, otherwise the module dirs will be duplicated, and that's the root cause of the ModuleLoadingManagementTestCase failed tests

Reply:
Nice catch, fixed.